src/http/routes.cc

```diff
@@ -81,8 +81,11 @@ std::unique_ptr<reply> routes::exception_reply(std::exception_ptr eptr) {
 }
 
 future<std::unique_ptr<reply> > routes::handle(const sstring& path, std::unique_ptr<request> req, std::unique_ptr<reply> rep) {
-    handler_base* handler = get_handler(str2type(req->_method),
-            normalize_url(path), req->param);
+    handler_base* handler = _global_handler;
+    if(handler == nullptr) {
```

Comment:
I think it's strange that in addition to the "global handler" you **allow** to set per-method and URL handlers, and yet ignore them if the global handler is set. Wouldn't it make more sense to have a "default handler", and allow other routes to override it?

Reply:
The existing logic is: first check for exact URL match then check for matching-rules.

maybe you want to do it here